src/lib/tabs/tab-body.ts

```diff
@@ -81,19 +81,21 @@ export class MatTabBodyPortal extends _MatTabBodyPortalBaseClass implements OnIn
   ngOnInit(): void {
     if (this._host._isCenterPosition(this._host._position)) {
       this.attach(this._host._content);
-    } else {
-      this._centeringSub = this._host._beforeCentering.subscribe(() => {
-        this.attach(this._host._content);
-        this._centeringSub.unsubscribe();
-      });
     }
+    this._centeringSub = this._host._beforeCentering.subscribe((isCentering: boolean) => {
+      if (isCentering) {
+        if (!this.hasAttached()) {
+          this.attach(this._host._content);
+        }
+      } else {
+        this.detach();
+      }
+    });
   }
 
-  /** Clean up subscription if necessary. */
+  /** Clean up centering subscription. */
   ngOnDestroy(): void {
-    if (this._centeringSub && !this._centeringSub.closed) {
-      this._centeringSub.unsubscribe();
-    }
+    this._centeringSub.unsubscribe();
```

Comment:
It probably still need to check whether `_centeringSub` exists because some people, in their unit tests, destroy a component before calling `detectChanges`